src/formats/notion-api.ts

```diff
@@ -1934,57 +1981,79 @@ export class NotionAPIImporter extends FormatImporter {
 	}
 
 	/**
-	 * Check if a file should be skipped during import
-	 * This applies to BOTH incremental and full import modes
-	 * 
+	 * Check if a file should be skipped, updated, or created during import
+	 * Supports delta sync by comparing timestamps when incremental import is enabled
+	 *
 	 * @param filePath - Path to the file to check
 	 * @param notionId - Notion ID of the page being imported
+	 * @param pageLastEdited - Last edited timestamp from Notion API
 	 * @param ctx - Import context for reporting
-	 * @returns true if file should be skipped, false otherwise
+	 * @returns Object indicating action: 'skip', 'update', or 'create'
 	 */
-	private async shouldSkipExistingFile(
+	private async shouldSkipOrUpdateFile(
 		filePath: string,
 		notionId: string,
+		pageLastEdited: string,
 		ctx: ImportContext
-	): Promise<boolean> {
+	): Promise<{ action: 'skip' | 'update' | 'create' }> {
```

Comment:
This doesn't need to be an object.

```suggestion
	): Promise<'skip' | 'update' | 'create'> {
```